### PR TITLE
Mark 3.1.0 as released

### PR DESCRIPTION
**Pull request by sbc100**
(should land after https://github.com/emscripten-core/emsdk/pull/955)

### CODE REVIEW

**Comment by truboxl** [comment]
The version change is not included in the 3.1.0 tag. Running emcc -v gets 3.0.2-git.
https://github.com/emscripten-core/emscripten/compare/3.1.0...main

**Comment by sbc100** [comment]
This is a side effect of how the versioning system in emscripten works.   When you run `emcc -v` from the git checkout out reports the current version + 1.. so in this case 3.0.2 which was expected to be the next release.

However since 3.0.2 was never released and we instead went to 3.1.0 if you happen to checkout the 3.1.0 tag it will report 3.0.2.    Normally this isn't an issue since 3.0.2 would normally follow 3.0.1.   This also doesn't effect emsdk users or git users who use tip-of-tree.   

See https://github.com/emscripten-core/emscripten/blob/main/docs/process.md#release-processes for a more detailed explanation. 

In short the version skew that you see here should be rare.. is it effecting you negatively in some way?

**Comment by truboxl** [comment]
I am responsible for packaging emscripten for Android. I usually git checkout the tags for version packaging. I believe other downstream package maintainers also faces this if they use git checkout (and do some sanity check). So far the tags match emcc -v up until this point.

So the question now is should downstream package maintainers package 3.1.0 tag as 3.1.0 or 3.0.2? Looking at commits between this PR and the tag reveals left out 2 changes that looks more like rename variables...
https://github.com/emscripten-core/emscripten/compare/3.1.0...57f35eefdc1ec865526804fbf28c81cbd7f28e6d

Doesnt look like much people picking up 3.1.0 so nobody questioning.
https://repology.org/project/emscripten/versions

**Comment by sbc100** [comment]
My suggestion would be to modify the `emscripten-version.txt` file as part of your packaging process.   `-git` -suffix that we keep there should probably also be removed when you do packaging.

In most cases the version numbers should match, and we can try to avoid this kind of thing going forward by updating the version number before making a release, but in general you probably want to use the tag name to set the `emscripten-version.txt`.    We do this when we do our own packaging as part of emsdk:  https://github.com/emscripten-core/emsdk/blob/060b48f0c69827f95c8433ee965656babf953e82/emsdk.py#L2039-L2047

**Comment by sbc100** [comment]
The tagged versions are what get releases as part of emsdk.
